Integrate slide sets with main website

crswong888 opened the pull request:
@aeslaughter @loganharbour we need to make sure that Civet does not build the slide sets separate from the main website if we merge this. We may also consider a new testing strategy.

(closes #18519)

crswong888 committed:
Implemented an 'alternative' setting for automatic links that serves as a fallback hyperlink should the original file fail to be found.

(refs #18519)

crswong888 committed:
unit tests and documentation for the AutoLink 'alternative' setting

(refs #18519)

crswong888 committed:
Configured MultiApps presentation to be built alongside the main website

(refs #18519)

crswong888 committed:
This is a fix for a bug that went unnoticed in #18211 - allowing duplicate directories in multiconfig builds creates a conflict when rendering breadcrumbs. And because of the way the _addBreadcrumbs() method invokes findPage(), simply deleting redundant Directory objects doesn't prevent the problem. It seems the only solution is to bypass the exception raised in findPage() for this very special case.

(refs #18519)

crswong888 committed:
Configured Code Verification presentation to be built alongside the main website

(refs #18519)

crswong888 committed:
Implemented a 'recursive' setting for the outline command to avoid including headers from the verification presentation when it is stored in the same directory as the website version in a multiconfig build

(refs #15366 #18519)

crswong888 committed:
adding a different background image to all slide sets as requested by @aeslaughter. I think some of the slides look janky with their content running into the high contrast INL footer, and I suggest we try a different design.

crswong888 committed:
Configuring slide sets to use the serial executioner since only a single page object is translated for reveal renderers and therefore do not benefit from multiprocessing and waste memory using it

(refs #18519)

crswong888 committed:
Created an alternative landing page for links to content that wasn't built.

On this page, I wanted to use an icon in the header so I made a new class to get the CSS right. Also, in order to use 'key=value' pairs for inline commands within headers, I had to modify the regex.

(refs #18519)

crswong888 committed:
modified automatic links to and fro slide sets to use the new 'alternative' setting

(refs #18519)

crswong888 committed:
simplifying the API needed for the live server

(refs #18211)

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit def87c9916e53ad08034f93db8fd82fb42578a85
+Subproject commit 6ebb480e2eb41cbbd62c330685e8be885d0ab276
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/795304/) on d473cb4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/18586/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Modules Parallel](https://civet.inl.gov/job/795313/) on d473cb4 : invalidated by @dschwen

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit def87c9916e53ad08034f93db8fd82fb42578a85
+Subproject commit 72362ac360dc1d424b54cc4bcf65e8108689d6ab
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

aeslaughter commented:
You have a size issue with the background?
https://mooseframework.inl.gov/docs/PRs/18586/site/getting_started/examples_and_tutorials/tutorial02_multiapps/presentation/index.html#/2/1

crswong888 committed:
styling to make reveal elements look sharp against high-contrast backgrounds

I also did some tweaking of HTML video elements, including properly enabling the "poster" attribute

(refs #18519)

crswong888 commented:
@aeslaughter The head commit here is my proposed solution to the reveal styling issues. Mainly, I'm adding box shadows to make elements like images, videos, tables, and bibliographies pop out more against the INL background.

I've glanced at every slide of every workshop, and I think everything looks pretty good. I don't want to take control of too much here, i.e., change global settings to conform to one type of background.
